From 5d708dfefe546251ff9aea533652377cb4530f50 Mon Sep 17 00:00:00 2001
From: Lukasz Majewski <lukma@denx.de>
Date: Tue, 1 Aug 2023 15:09:32 +0200
Subject: [PATCH 13/15] dts: Exclude common Kostal device tree description to
 new dtsi file

As it was easier to fix the DTS - changes are applied to
imx6-kie-inverter.dtsi file:

- Fix the PWM backlight by adding #pwm-cells = <2>; to &pwm6 node (5.10.188)

- Remove commented out portions of the original DTS (display, backlight) as
  those are implemented by Kostal

Signed-off-by: Lukasz Majewski <lukma@denx.de>
---
 arch/arm/boot/dts/imx6-kie-inverter.dtsi   | 415 ++++++++++++++++++++
 arch/arm/boot/dts/imx6ull-kie-inverter.dts | 433 +--------------------
 2 files changed, 416 insertions(+), 432 deletions(-)
 create mode 100644 arch/arm/boot/dts/imx6-kie-inverter.dtsi

diff --git a/arch/arm/boot/dts/imx6-kie-inverter.dtsi b/arch/arm/boot/dts/imx6-kie-inverter.dtsi
new file mode 100644
index 000000000000..9c464c0789e9
--- /dev/null
+++ b/arch/arm/boot/dts/imx6-kie-inverter.dtsi
@@ -0,0 +1,415 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2019 DENX Software Engineering GmbH
+ * Author: Parthiban Nallathambi <pn@denx.de>
+ */
+
+#include "imx6ull-kie-inverter-display.dtsi"
+#include "imx6ull-kie-inverter-leds.dtsi"
+#include "imx6ull-kie-inverter-keys.dtsi"
+/ {
+	aliases {
+		u-boot,dm-pre-reloc;
+		mmc0 = &usdhc2;
+		usb0 = &usbotg1;
+		ethernet0 = &fec2;
+	};
+
+	backlight {
+		compatible = "pwm-backlight";
+		pinctrl-names = "default";
+		pwms = <&pwm6 0 30000>;
+		brightness-levels = <0 5 13 23 31 48 66 94 148 255>;
+		default-brightness-level = <8>;
+		status = "okay";
+	};
+
+	chosen {
+		stdout-path = &uart1;
+	};
+
+	memory {
+		reg = <0x80000000 0x20000000>;
+	};
+
+	regulators: regulators {
+		compatible = "simple-bus";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		reg_usbotg2_vbus: usbotg2-vbus@0 {
+			compatible = "regulator-fixed";
+			pinctrl-names = "default";
+			pinctrl-0 = <&pinctrl_usb_otg2_vbus>;
+			regulator-name = "usb_otg2_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio1 4 GPIO_ACTIVE_HIGH>;
+			enable-active-high;
+		};
+
+		VDD_in: regulator@1 {
+			compatible = "regulator-fixed";
+			regulator-name = "VDD_in";
+			regulator-min-microvolt = <12000000>;
+			regulator-max-microvolt = <12000000>;
+			regulator-always-on;
+			regulator-boot-on;
+		};
+
+		vcc_5v: regulator@2 {
+			compatible = "regulator-fixed";
+			regulator-name = "+5V";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			regulator-always-on;
+			vin-supply = <&VDD_in>;
+		};
+
+		reg_3v3: regulator@4 {
+			compatible = "regulator-fixed";
+			regulator-name = "3V3";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			regulator-always-on;
+			vin-supply = <&vcc_5v>;
+		};
+
+		reg_1v8: regulator@5 {
+			compatible = "regulator-fixed";
+			regulator-name = "1V8";
+			regulator-min-microvolt = <1800000>;
+			regulator-max-microvolt = <1800000>;
+			regulator-always-on;
+			vin-supply = <&vcc_5v>;
+		};
+
+		reg_1v35: regulator@6 {
+			compatible = "regulator-fixed";
+			regulator-name = "1V35";
+			regulator-min-microvolt = <1350000>;
+			regulator-max-microvolt = <1350000>;
+			regulator-always-on;
+			vin-supply = <&vcc_5v>;
+		};
+	};
+
+	wifi_pwrseq: sdio-pwrseq {
+		compatible = "mmc-pwrseq-simple";
+		pinctrl-names = "default";
+		pinctrl-0 = <&sdio_wifi_pwrseq>;
+		reset-gpios = <&gpio1 1 GPIO_ACTIVE_LOW>;
+		post-power-on-delay-ms = <2>;
+	};
+
+	watchdog: watchdog {
+		compatible = "linux,wdt-gpio";
+		gpios = <&gpio1 5 GPIO_ACTIVE_HIGH>;
+		pinctrl-0 = <&pinctrl_ext_wdog>;
+		hw_algo = "toggle";
+		hw_margin_ms = <1600>;
+		always-running;
+	};
+};
+
+&can1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_flexcan1>;
+	xceiver-supply = <&vcc_5v>;
+	status = "okay";
+};
+
+&pwm6 {
+	#pwm-cells = <2>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm6>;
+	status = "okay";
+};
+
+&fec2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_enet2>;
+	phy-mode = "rmii";
+	phy-reset-gpios = <&gpio3 22 GPIO_ACTIVE_LOW>;
+	phy-reset-duration = <10>;
+	phy-reset-post-delay = <20>;
+	phy-supply = <&reg_3v3>;
+
+	clocks = <&clks IMX6UL_CLK_ENET>,
+		 <&clks IMX6UL_CLK_ENET_AHB>,
+		 <&clks IMX6UL_CLK_ENET_PTP>;
+	clock-names = "ipg", "ahb", "ptp";
+
+	status = "okay";
+
+	fixed-link {
+		speed = <100>;
+		full-duplex;
+	};
+
+	mdio {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		switch@10 {
+			compatible = "marvell,mv88e6250";
+			reg = <0x10>;
+			interrupt-parent = <&gpio3>;
+			interrupts = <23 IRQ_TYPE_LEVEL_LOW>;
+			interrupt-controller;
+			#interrupt-cells = <2>;
+
+			ports {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				port@0 {
+					reg = <0>;
+					label = "lan1";
+				};
+
+				port@1 {
+					reg = <1>;
+					label = "lan2";
+				};
+
+				port@6 {
+					reg = <6>;
+					label = "cpu";
+					phy-mode = "rmii";
+					ethernet = <&fec2>;
+
+					fixed-link {
+						   speed = <100>;
+						   full-duplex;
+					};
+				};
+			};
+		};
+	};
+};
+
+&i2c1 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default", "gpio";
+	pinctrl-0 = <&pinctrl_i2c1>;
+	status = "okay";
+
+	pca8574: gpio@20 {
+		compatible = "nxp,pca8574";
+		reg = <0x20>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		interrupt-parent = <&gpio1>;
+		interrupts = <8 IRQ_TYPE_LEVEL_LOW>;
+	};
+};
+
+
+
+&uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart1>;
+	status = "okay";
+};
+
+&uart2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart2>;
+	uart-has-rtscts;
+	status = "okay";
+};
+
+&usbotg1 {
+	/* TODO Fixed regulator needed? */
+	disable-over-current;
+	dr_mode = "host";
+	status = "okay";
+};
+
+&usbotg2 {
+	vbus-supply = <&reg_usbotg2_vbus>;
+	disable-over-current;
+	dr_mode = "host";
+	status = "okay";
+};
+
+&usdhc1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc1>;
+	bus-width = <4>;
+	no-1-8-v;       /* force 3.3V VIO */
+	non-removable;
+	vmmc-supply = <&reg_3v3>;
+	keep-power-in-suspend;
+	mmc-pwrseq = <&wifi_pwrseq>;
+	max-frequency = <25000000>;
+	status = "okay";
+
+	wlan@1 {
+	       compatible = "marvell,sd8897";
+	       reg = <1>;
+	};
+};
+
+&usdhc2 {
+	u-boot,dm-spl;
+	u-boot,dm-pre-reloc;
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc2>;
+	pinctrl-1 = <&pinctrl_usdhc2_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc2_200mhz>;
+	bus-width = <8>;
+	non-removable;
+	keep-power-in-suspend;
+	status = "okay";
+};
+
+&iomuxc {
+	pinctrl-names = "default";
+
+	pinctrl_enet2: enet2grp {
+		fsl,pins = <
+			MX6UL_PAD_GPIO1_IO06__ENET2_MDIO	0x10010
+			MX6UL_PAD_GPIO1_IO07__ENET2_MDC		0x10010
+			MX6UL_PAD_ENET2_RX_EN__ENET2_RX_EN	0x1b0b0
+			MX6UL_PAD_ENET2_RX_ER__GPIO2_IO15       0x80000000
+			MX6UL_PAD_ENET2_RX_DATA0__ENET2_RDATA00	0x1b0b0
+			MX6UL_PAD_ENET2_RX_DATA1__ENET2_RDATA01	0x1b0b0
+			MX6UL_PAD_ENET2_TX_EN__ENET2_TX_EN	0x1b010
+			MX6UL_PAD_ENET2_TX_DATA0__ENET2_TDATA00	0x1b010
+			MX6UL_PAD_ENET2_TX_DATA1__ENET2_TDATA01	0x1b010
+			MX6UL_PAD_ENET2_TX_CLK__ENET2_REF_CLK2	0x4001b031
+			MX6UL_PAD_LCD_DATA17__GPIO3_IO22        0x80000000
+			MX6UL_PAD_LCD_DATA18__GPIO3_IO23        0x17059
+		>;
+	};
+
+	pinctrl_ext_wdog: extwdoggrp {
+		fsl,pins = <
+			MX6UL_PAD_GPIO1_IO05__GPIO1_IO05	0x1b0b0
+			/* TODO WDT_SEL, no support in kernel for GPIO */
+			MX6UL_PAD_GPIO1_IO03__GPIO1_IO03	0x1b0b0
+		>;
+	};
+
+	pinctrl_flexcan1: flexcan1grp {
+		fsl,pins = <
+			MX6UL_PAD_LCD_DATA08__FLEXCAN1_TX	0x1b020
+			MX6UL_PAD_LCD_DATA09__FLEXCAN1_RX	0x1b020
+			/* TODO STB - silent mode */
+			MX6UL_PAD_LCD_DATA10__GPIO3_IO15	0x1b0b0
+		>;
+	};
+
+	pinctrl_i2c1: i2cgrp {
+		fsl,pins = <
+			MX6UL_PAD_UART4_TX_DATA__I2C1_SCL	0x4001b8b0
+			MX6UL_PAD_UART4_RX_DATA__I2C1_SDA	0x4001b8b0
+			MX6UL_PAD_GPIO1_IO08__GPIO1_IO08	0x0b0b0
+		>;
+	};
+
+	sdio_wifi_pwrseq: wifipdngrp {
+		fsl,pins = <
+			/* WL_PDN */
+			MX6UL_PAD_GPIO1_IO01__GPIO1_IO01	0x10b0
+		>;
+	};
+
+	pinctrl_pwm6: pwm6grp {
+		fsl,pins = <
+			MX6UL_PAD_LCD_DATA19__PWM6_OUT		0x1b0b1
+		>;
+	};
+
+	pinctrl_uart1: uart1grp {
+		fsl,pins = <
+			MX6UL_PAD_UART1_TX_DATA__UART1_DCE_TX	0x1b0b1
+			MX6UL_PAD_UART1_RX_DATA__UART1_DCE_RX	0x1b0b1
+		>;
+	};
+
+	pinctrl_uart2: uart2grp {
+		fsl,pins = <
+			MX6UL_PAD_UART2_TX_DATA__UART2_DCE_TX	0x1b0b1
+			MX6UL_PAD_UART2_RX_DATA__UART2_DCE_RX	0x1b0b1
+			MX6UL_PAD_UART2_CTS_B__UART2_DCE_CTS	0x1b0b1
+			MX6UL_PAD_UART2_RTS_B__UART2_DCE_RTS	0x1b0b1
+		>;
+	};
+
+	pinctrl_usb_otg2_vbus: usbotg2vbusgrp {
+		fsl,pins = <
+			MX6UL_PAD_GPIO1_IO04__GPIO1_IO04	0x10b0
+		>;
+	};
+
+	pinctrl_usdhc1: usdhc1grp {
+		fsl,pins = <
+			MX6UL_PAD_SD1_CMD__USDHC1_CMD		0x170f9
+			MX6UL_PAD_SD1_CLK__USDHC1_CLK		0x100f9
+			MX6UL_PAD_SD1_DATA0__USDHC1_DATA0       0x170f9
+			MX6UL_PAD_SD1_DATA1__USDHC1_DATA1       0x170f9
+			MX6UL_PAD_SD1_DATA2__USDHC1_DATA2       0x170f9
+			MX6UL_PAD_SD1_DATA3__USDHC1_DATA3       0x170f9
+
+			/* WL_HOST_WAKE */
+			MX6UL_PAD_GPIO1_IO00__GPIO1_IO00	0x10b0
+
+			/* WL_SD_WAKE */
+			MX6UL_PAD_GPIO1_IO02__GPIO1_IO02	0x10b0
+
+			/* WL_GPIO0 */
+			MX6UL_PAD_LCD_DATA21__GPIO3_IO26	0x10b0
+		>;
+	};
+
+	pinctrl_usdhc2: usdhc2grp {
+		fsl,pins = <
+			MX6UL_PAD_NAND_WE_B__USDHC2_CMD		0x17059
+			MX6UL_PAD_NAND_RE_B__USDHC2_CLK		0x10059
+			MX6UL_PAD_NAND_DATA00__USDHC2_DATA0	0x17059
+			MX6UL_PAD_NAND_DATA01__USDHC2_DATA1	0x17059
+			MX6UL_PAD_NAND_DATA02__USDHC2_DATA2	0x17059
+			MX6UL_PAD_NAND_DATA03__USDHC2_DATA3	0x17059
+			MX6UL_PAD_NAND_DATA04__USDHC2_DATA4	0x17059
+			MX6UL_PAD_NAND_DATA05__USDHC2_DATA5	0x17059
+			MX6UL_PAD_NAND_DATA06__USDHC2_DATA6	0x17059
+			MX6UL_PAD_NAND_DATA07__USDHC2_DATA7	0x17059
+			MX6UL_PAD_NAND_ALE__USDHC2_RESET_B	0x17059
+		>;
+	};
+
+	pinctrl_usdhc2_100mhz: usdhc2grp100mhz {
+		fsl,pins = <
+			MX6UL_PAD_NAND_WE_B__USDHC2_CMD		0x170b9
+			MX6UL_PAD_NAND_RE_B__USDHC2_CLK		0x100b9
+			MX6UL_PAD_NAND_DATA00__USDHC2_DATA0	0x170b9
+			MX6UL_PAD_NAND_DATA01__USDHC2_DATA1	0x170b9
+			MX6UL_PAD_NAND_DATA02__USDHC2_DATA2	0x170b9
+			MX6UL_PAD_NAND_DATA03__USDHC2_DATA3	0x170b9
+			MX6UL_PAD_NAND_DATA04__USDHC2_DATA4	0x170b9
+			MX6UL_PAD_NAND_DATA05__USDHC2_DATA5	0x170b9
+			MX6UL_PAD_NAND_DATA06__USDHC2_DATA6	0x170b9
+			MX6UL_PAD_NAND_DATA07__USDHC2_DATA7	0x170b9
+			MX6UL_PAD_NAND_ALE__USDHC2_RESET_B	0x170b9
+		>;
+	};
+
+	pinctrl_usdhc2_200mhz: usdhc2grp200mhz {
+		fsl,pins = <
+			MX6UL_PAD_NAND_WE_B__USDHC2_CMD		0x170f9
+			MX6UL_PAD_NAND_RE_B__USDHC2_CLK		0x100f9
+			MX6UL_PAD_NAND_DATA00__USDHC2_DATA0	0x170f9
+			MX6UL_PAD_NAND_DATA01__USDHC2_DATA1	0x170f9
+			MX6UL_PAD_NAND_DATA02__USDHC2_DATA2	0x170f9
+			MX6UL_PAD_NAND_DATA03__USDHC2_DATA3	0x170f9
+			MX6UL_PAD_NAND_DATA04__USDHC2_DATA4	0x170f9
+			MX6UL_PAD_NAND_DATA05__USDHC2_DATA5	0x170f9
+			MX6UL_PAD_NAND_DATA06__USDHC2_DATA6	0x170f9
+			MX6UL_PAD_NAND_DATA07__USDHC2_DATA7	0x170f9
+			MX6UL_PAD_NAND_ALE__USDHC2_RESET_B	0x170f9
+		>;
+	};
+};
diff --git a/arch/arm/boot/dts/imx6ull-kie-inverter.dts b/arch/arm/boot/dts/imx6ull-kie-inverter.dts
index 215e67187c84..782e8206cba6 100644
--- a/arch/arm/boot/dts/imx6ull-kie-inverter.dts
+++ b/arch/arm/boot/dts/imx6ull-kie-inverter.dts
@@ -6,439 +6,8 @@
 /dts-v1/;
 
 #include "imx6ull.dtsi"
-#include "imx6ull-kie-inverter-display.dtsi"
-#include "imx6ull-kie-inverter-leds.dtsi"
-#include "imx6ull-kie-inverter-keys.dtsi"
+#include "imx6-kie-inverter.dtsi"
 / {
 	model = "Kostal KIE imx6 Ultra Lite Inverter";
 	compatible = "kostal,imx6ull-inverter", "fsl,imx6ull";
-
-	aliases {
-		u-boot,dm-pre-reloc;
-		mmc0 = &usdhc2;
-		usb0 = &usbotg1;
-		ethernet0 = &fec2;
-	};
-
-	backlight: backlight {
-		compatible = "pwm-backlight";
-		pinctrl-names = "default";
-		pwms = <&pwm6 0 30000>;
-		brightness-levels = <0 5 13 23 31 48 66 94 148 255>;
-		default-brightness-level = <8>;
-		status = "okay";
-	};
-
-	chosen {
-		stdout-path = &uart1;
-	};
-
-	memory {
-		reg = <0x80000000 0x20000000>;
-	};
-
-	regulators: regulators {
-		compatible = "simple-bus";
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		reg_usbotg2_vbus: usbotg2-vbus@0 {
-			compatible = "regulator-fixed";
-			pinctrl-names = "default";
-			pinctrl-0 = <&pinctrl_usb_otg2_vbus>;
-			regulator-name = "usb_otg2_vbus";
-			regulator-min-microvolt = <5000000>;
-			regulator-max-microvolt = <5000000>;
-			gpio = <&gpio1 4 GPIO_ACTIVE_HIGH>;
-			enable-active-high;
-		};
-
-		VDD_in: regulator@1 {
-			compatible = "regulator-fixed";
-			regulator-name = "VDD_in";
-			regulator-min-microvolt = <12000000>;
-			regulator-max-microvolt = <12000000>;
-			regulator-always-on;
-			regulator-boot-on;
-		};
-
-		vcc_5v: regulator@2 {
-			compatible = "regulator-fixed";
-			regulator-name = "+5V";
-			regulator-min-microvolt = <5000000>;
-			regulator-max-microvolt = <5000000>;
-			regulator-always-on;
-			vin-supply = <&VDD_in>;
-		};
-
-		reg_3v3: regulator@4 {
-			compatible = "regulator-fixed";
-			regulator-name = "3V3";
-			regulator-min-microvolt = <3300000>;
-			regulator-max-microvolt = <3300000>;
-			regulator-always-on;
-			vin-supply = <&vcc_5v>;
-		};
-
-		reg_1v8: regulator@5 {
-			compatible = "regulator-fixed";
-			regulator-name = "1V8";
-			regulator-min-microvolt = <1800000>;
-			regulator-max-microvolt = <1800000>;
-			regulator-always-on;
-			vin-supply = <&vcc_5v>;
-		};
-
-		reg_1v35: regulator@6 {
-			compatible = "regulator-fixed";
-			regulator-name = "1V35";
-			regulator-min-microvolt = <1350000>;
-			regulator-max-microvolt = <1350000>;
-			regulator-always-on;
-			vin-supply = <&vcc_5v>;
-		};
-
-
-/*
-		vbus_otg1: regulator@3 {
-			compatible = "regulator-fixed";
-			reg = <3>;
-			regulator-name = "vbus_otg1";
-			regulator-min-microvolt = <5000000>;
-			regulator-max-microvolt = <5000000>;
-		};
-*/
-	};
-
-
-/*
-	panel {
-		compatible = "sitronix,st7701";
-		backlight = <&backlight>;
-
-		port {
-			panel_in: endpoint {
-				remote-endpoint = <&display_out>;
-			};
-		};
-	};
-*/
-
-	wifi_pwrseq: sdio-pwrseq {
-		compatible = "mmc-pwrseq-simple";
-		pinctrl-names = "default";
-		pinctrl-0 = <&sdio_wifi_pwrseq>;
-		reset-gpios = <&gpio1 1 GPIO_ACTIVE_LOW>;
-		post-power-on-delay-ms = <2>;
-	};
-
-	watchdog: watchdog {
-		compatible = "linux,wdt-gpio";
-		gpios = <&gpio1 5 GPIO_ACTIVE_HIGH>;
-		pinctrl-0 = <&pinctrl_ext_wdog>;
-		hw_algo = "toggle";
-		hw_margin_ms = <1600>;
-		always-running;
-	};
-};
-
-&can1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_flexcan1>;
-	xceiver-supply = <&vcc_5v>;
-	status = "okay";
-};
-
-&pwm6 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_pwm6>;
-	status = "okay";
-};
-
-&fec2 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_enet2>;
-	phy-mode = "rmii";
-	phy-reset-gpios = <&gpio3 22 GPIO_ACTIVE_LOW>;
-	phy-reset-duration = <10>;
-	phy-reset-post-delay = <20>;
-	phy-supply = <&reg_3v3>;
-
-	clocks = <&clks IMX6UL_CLK_ENET>,
-		 <&clks IMX6UL_CLK_ENET_AHB>,
-		 <&clks IMX6UL_CLK_ENET_PTP>;
-	clock-names = "ipg", "ahb", "ptp";
-
-	status = "okay";
-
-	fixed-link {
-		speed = <100>;
-		full-duplex;
-	};
-
-	mdio {
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		switch@10 {
-			compatible = "marvell,mv88e6250";
-			reg = <0x10>;
-			interrupt-parent = <&gpio3>;
-			interrupts = <23 IRQ_TYPE_LEVEL_LOW>;
-			interrupt-controller;
-			#interrupt-cells = <2>;
-
-			ports {
-				#address-cells = <1>;
-				#size-cells = <0>;
-
-				port@0 {
-					reg = <0>;
-					label = "lan1";
-				};
-
-				port@1 {
-					reg = <1>;
-					label = "lan2";
-				};
-
-				port@6 {
-					reg = <6>;
-					label = "cpu";
-					phy-mode = "rmii";
-					ethernet = <&fec2>;
-
-					fixed-link {
-						   speed = <100>;
-						   full-duplex;
-					};
-				};
-			};
-		};
-	};
-};
-
-&i2c1 {
-	clock-frequency = <100000>;
-	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl_i2c1>;
-	status = "okay";
-
-	pca8574: gpio@20 {
-		compatible = "nxp,pca8574";
-		reg = <0x20>;
-		gpio-controller;
-		#gpio-cells = <2>;
-		interrupt-parent = <&gpio1>;
-		interrupts = <8 IRQ_TYPE_LEVEL_LOW>;
-	};
-};
-
-
-
-&uart1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart1>;
-	status = "okay";
-};
-
-&uart2 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart2>;
-	uart-has-rtscts;
-	status = "okay";
-};
-
-&usbotg1 {
-	/* TODO Fixed regulator needed? */
-	disable-over-current;
-	dr_mode = "host";
-	status = "okay";
-};
-
-&usbotg2 {
-	vbus-supply = <&reg_usbotg2_vbus>;
-	disable-over-current;
-	dr_mode = "host";
-	status = "okay";
-};
-
-&usdhc1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_usdhc1>;
-	bus-width = <4>;
-	no-1-8-v;       /* force 3.3V VIO */
-	non-removable;
-	vmmc-supply = <&reg_3v3>;
-	keep-power-in-suspend;
-	mmc-pwrseq = <&wifi_pwrseq>;
-	max-frequency = <25000000>;
-	status = "okay";
-
-	wlan@1 {
-	       compatible = "marvell,sd8897";
-	       reg = <1>;
-	};
-};
-
-&usdhc2 {
-	u-boot,dm-spl;
-	u-boot,dm-pre-reloc;
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
-	pinctrl-0 = <&pinctrl_usdhc2>;
-	pinctrl-1 = <&pinctrl_usdhc2_100mhz>;
-	pinctrl-2 = <&pinctrl_usdhc2_200mhz>;
-	bus-width = <8>;
-	non-removable;
-	keep-power-in-suspend;
-	status = "okay";
-};
-
-&iomuxc {
-	pinctrl-names = "default";
-
-	pinctrl_enet2: enet2grp {
-		fsl,pins = <
-			MX6UL_PAD_GPIO1_IO06__ENET2_MDIO	0x10010
-			MX6UL_PAD_GPIO1_IO07__ENET2_MDC		0x10010
-			MX6UL_PAD_ENET2_RX_EN__ENET2_RX_EN	0x1b0b0
-			MX6UL_PAD_ENET2_RX_ER__GPIO2_IO15       0x80000000
-			MX6UL_PAD_ENET2_RX_DATA0__ENET2_RDATA00	0x1b0b0
-			MX6UL_PAD_ENET2_RX_DATA1__ENET2_RDATA01	0x1b0b0
-			MX6UL_PAD_ENET2_TX_EN__ENET2_TX_EN	0x1b010
-			MX6UL_PAD_ENET2_TX_DATA0__ENET2_TDATA00	0x1b010
-			MX6UL_PAD_ENET2_TX_DATA1__ENET2_TDATA01	0x1b010
-			MX6UL_PAD_ENET2_TX_CLK__ENET2_REF_CLK2	0x4001b031
-			MX6UL_PAD_LCD_DATA17__GPIO3_IO22        0x80000000
-			MX6UL_PAD_LCD_DATA18__GPIO3_IO23        0x17059
-		>;
-	};
-
-	pinctrl_ext_wdog: extwdoggrp {
-		fsl,pins = <
-			MX6UL_PAD_GPIO1_IO05__GPIO1_IO05	0x1b0b0
-			/* TODO WDT_SEL, no support in kernel for GPIO */
-			MX6UL_PAD_GPIO1_IO03__GPIO1_IO03	0x1b0b0
-		>;
-	};
-
-	pinctrl_flexcan1: flexcan1grp {
-		fsl,pins = <
-			MX6UL_PAD_LCD_DATA08__FLEXCAN1_TX	0x1b020
-			MX6UL_PAD_LCD_DATA09__FLEXCAN1_RX	0x1b020
-			/* TODO STB - silent mode */
-			MX6UL_PAD_LCD_DATA10__GPIO3_IO15	0x1b0b0
-		>;
-	};
-
-	pinctrl_i2c1: i2cgrp {
-		fsl,pins = <
-			MX6UL_PAD_UART4_TX_DATA__I2C1_SCL	0x4001b8b0
-			MX6UL_PAD_UART4_RX_DATA__I2C1_SDA	0x4001b8b0
-			MX6UL_PAD_GPIO1_IO08__GPIO1_IO08	0x0b0b0
-		>;
-	};
-
-	sdio_wifi_pwrseq: wifipdngrp {
-		fsl,pins = <
-			/* WL_PDN */
-			MX6UL_PAD_GPIO1_IO01__GPIO1_IO01	0x10b0
-		>;
-	};
-
-	pinctrl_pwm6: pwm6grp {
-		fsl,pins = <
-			MX6UL_PAD_LCD_DATA19__PWM6_OUT		0x1b0b1
-		>;
-	};
-
-	pinctrl_uart1: uart1grp {
-		fsl,pins = <
-			MX6UL_PAD_UART1_TX_DATA__UART1_DCE_TX	0x1b0b1
-			MX6UL_PAD_UART1_RX_DATA__UART1_DCE_RX	0x1b0b1
-		>;
-	};
-
-	pinctrl_uart2: uart2grp {
-		fsl,pins = <
-			MX6UL_PAD_UART2_TX_DATA__UART2_DCE_TX	0x1b0b1
-			MX6UL_PAD_UART2_RX_DATA__UART2_DCE_RX	0x1b0b1
-			MX6UL_PAD_UART2_CTS_B__UART2_DCE_CTS	0x1b0b1
-			MX6UL_PAD_UART2_RTS_B__UART2_DCE_RTS	0x1b0b1
-		>;
-	};
-
-	pinctrl_usb_otg2_vbus: usbotg2vbusgrp {
-		fsl,pins = <
-			MX6UL_PAD_GPIO1_IO04__GPIO1_IO04	0x10b0
-		>;
-	};
-
-	pinctrl_usdhc1: usdhc1grp {
-		fsl,pins = <
-			MX6UL_PAD_SD1_CMD__USDHC1_CMD		0x170f9
-			MX6UL_PAD_SD1_CLK__USDHC1_CLK		0x100f9
-			MX6UL_PAD_SD1_DATA0__USDHC1_DATA0       0x170f9
-			MX6UL_PAD_SD1_DATA1__USDHC1_DATA1       0x170f9
-			MX6UL_PAD_SD1_DATA2__USDHC1_DATA2       0x170f9
-			MX6UL_PAD_SD1_DATA3__USDHC1_DATA3       0x170f9
-
-			/* WL_HOST_WAKE */
-			MX6UL_PAD_GPIO1_IO00__GPIO1_IO00	0x10b0
-
-			/* WL_SD_WAKE */
-			MX6UL_PAD_GPIO1_IO02__GPIO1_IO02	0x10b0
-
-			/* WL_GPIO0 */
-			MX6UL_PAD_LCD_DATA21__GPIO3_IO26	0x10b0
-		>;
-	};
-
-	pinctrl_usdhc2: usdhc2grp {
-		fsl,pins = <
-			MX6UL_PAD_NAND_WE_B__USDHC2_CMD		0x17059
-			MX6UL_PAD_NAND_RE_B__USDHC2_CLK		0x10059
-			MX6UL_PAD_NAND_DATA00__USDHC2_DATA0	0x17059
-			MX6UL_PAD_NAND_DATA01__USDHC2_DATA1	0x17059
-			MX6UL_PAD_NAND_DATA02__USDHC2_DATA2	0x17059
-			MX6UL_PAD_NAND_DATA03__USDHC2_DATA3	0x17059
-			MX6UL_PAD_NAND_DATA04__USDHC2_DATA4	0x17059
-			MX6UL_PAD_NAND_DATA05__USDHC2_DATA5	0x17059
-			MX6UL_PAD_NAND_DATA06__USDHC2_DATA6	0x17059
-			MX6UL_PAD_NAND_DATA07__USDHC2_DATA7	0x17059
-			MX6UL_PAD_NAND_ALE__USDHC2_RESET_B	0x17059
-		>;
-	};
-
-	pinctrl_usdhc2_100mhz: usdhc2grp100mhz {
-		fsl,pins = <
-			MX6UL_PAD_NAND_WE_B__USDHC2_CMD		0x170b9
-			MX6UL_PAD_NAND_RE_B__USDHC2_CLK		0x100b9
-			MX6UL_PAD_NAND_DATA00__USDHC2_DATA0	0x170b9
-			MX6UL_PAD_NAND_DATA01__USDHC2_DATA1	0x170b9
-			MX6UL_PAD_NAND_DATA02__USDHC2_DATA2	0x170b9
-			MX6UL_PAD_NAND_DATA03__USDHC2_DATA3	0x170b9
-			MX6UL_PAD_NAND_DATA04__USDHC2_DATA4	0x170b9
-			MX6UL_PAD_NAND_DATA05__USDHC2_DATA5	0x170b9
-			MX6UL_PAD_NAND_DATA06__USDHC2_DATA6	0x170b9
-			MX6UL_PAD_NAND_DATA07__USDHC2_DATA7	0x170b9
-			MX6UL_PAD_NAND_ALE__USDHC2_RESET_B	0x170b9
-		>;
-	};
-
-	pinctrl_usdhc2_200mhz: usdhc2grp200mhz {
-		fsl,pins = <
-			MX6UL_PAD_NAND_WE_B__USDHC2_CMD		0x170f9
-			MX6UL_PAD_NAND_RE_B__USDHC2_CLK		0x100f9
-			MX6UL_PAD_NAND_DATA00__USDHC2_DATA0	0x170f9
-			MX6UL_PAD_NAND_DATA01__USDHC2_DATA1	0x170f9
-			MX6UL_PAD_NAND_DATA02__USDHC2_DATA2	0x170f9
-			MX6UL_PAD_NAND_DATA03__USDHC2_DATA3	0x170f9
-			MX6UL_PAD_NAND_DATA04__USDHC2_DATA4	0x170f9
-			MX6UL_PAD_NAND_DATA05__USDHC2_DATA5	0x170f9
-			MX6UL_PAD_NAND_DATA06__USDHC2_DATA6	0x170f9
-			MX6UL_PAD_NAND_DATA07__USDHC2_DATA7	0x170f9
-			MX6UL_PAD_NAND_ALE__USDHC2_RESET_B	0x170f9
-		>;
-	};
 };
-- 
2.37.2

